Automattic/wp-calypso
# Limited Global Styles: Fix modal layout

Fixes https://github.com/Automattic/wp-calypso/issues/76089

## Proposed Changes

As of [Gutenberg 15.2](https://github.com/WordPress/gutenberg/pull/47426), the `Modal` component no longer renders the passed React nodes as direct children of `.components-modal__content`, effectively breaking the flex layout used by our Limited Global Styles gating modal.

To fix that, we now use a wrapper with a custom class so we can build a flex layout without relying on how Gutenberg renders the `Modal` component.

Before | After
--- | ---
![image](https://user-images.githubusercontent.com/1233880/233990789-a29b530f-e150-4d1e-bfa9-8ea417fbda49.png) | <img width="786" alt="Screenshot 2023-04-24 at 13 58 47" src="https://user-images.githubusercontent.com/1233880/233990823-80c34b7c-cf49-4bc6-9a42-c3ab2612127b.png">


## Testing Instructions

- Apply these changes to your sandbox
- Create a new site and sandbox it
- Go to the site editor and open the styles sidebar
- Make sure the modal has the proper layout
